### ISSUE TITLE
🐛 In side-by-side mode, tabs mess up alignment of plus side

### ISSUE THREAD

**Issue by dhduvall** [dandavison/delta#750]
If there are raw tabs in the output in side-by-side mode, then the second (plus) column isn't well aligned:
<img width="768" alt="Screen Shot (bad)" src="https://user-images.githubusercontent.com/7074165/139111945-36bd505b-39fe-4875-ae80-c776db6b1bc4.png">
It should instead look like this:
<img width="765" alt="Screen Shot (good)" src="https://user-images.githubusercontent.com/7074165/139112111-7ae4041b-5f61-49c4-91c1-3d146c4eebcf.png">
This is related to #729, but comes from a different part of the code ([`ansi::measure_text_width()`](https://github.com/dandavison/delta/blob/081eb60299a0f75670f8397e58ed6c05417747b4/src/ansi/mod.rs#L21-L24)).

**Comment by th1000s** [comment]
Note that using `--tabs` like this is documented to possibly not work. The side-by-side mode could exit with an error here, or assume that however uses it like that knows what they are doing, such as their own postprocessing.

> Use --tabs=0 to pass tab characters through directly,
> but note that in that case delta will calculate line widths assuming tabs occupy one character's width on
> the screen: **if your terminal renders tabs as more than than one character wide** then delta's output will **look
> incorrect**.

**Comment by dhduvall** [comment]
Right; there's no way to guarantee that it'll do the right thing. But it can do better than it does currently, if we make this assumption about terminal tab width. I'd be curious to see what terminal emulators set the default tab width to anything other than 8 columns, but as long as the major ones do, then assuming 8 is probably fine (or it could be configurable for people who go that far away from the norm). Maybe I'm wrong about the prevalence of 8-column tabs?

I think that `--tabs=0` is a convenient workaround for being unable to see changes between tabs and spaces because tabs have been replaced before the highlighting happens. At least, that's why I started experimenting with it. So if/when that gets fixed, there won't be a need to do anything nice here. But until then ... (I start a new job tomorrow, so I may be slow in making that happen).

Anyway, I'll defer to you and Dan on this one.

**Comment by th1000s** [comment]
I learned a new weird terminal command: `tabs`

This can change the width to e.g. 1 `tabs -1`, but can also set completely arbitrary tabstops: `tabs 2,+2,+5,+11,+12,+1,+4`. But I don't think anyone seriously uses that, I also think that most terminals do use 8. Still, lets fix general tab behavior in the other PR first.
